micro-rdk/src/common/registry.rs

```diff
@@ -202,6 +209,18 @@ impl ComponentRegistry {
             dependencies: dependency_func_map,
         }
     }
+    pub fn register_button(
+        &mut self,
+        model: impl Into<String>,
+        constructor: &'static ButtonConstructor,
+    ) -> Result<(), RegistryError> {
+        let model = model.into();
+        if self.buttons.contains_key(&model) {
+            return Err(RegistryError::ModelAlreadyRegistered(model));
+        }
+        let _ = self.buttons.insert(model, constructor);
```

Comment:
this is the same as everything else, but not sure why we ignore the result in the first place